cartography/intel/semgrep/dependencies.py

```diff
@@ -157,39 +180,50 @@ def load_dependencies(
 @timeit
 def cleanup(
     neo4j_session: neo4j.Session,
+    dependency_schema: Callable,
     common_job_parameters: Dict[str, Any],
 ) -> None:
-    logger.info("Running Semgrep Go Library cleanup job.")
-    go_libraries_cleanup_job = GraphJob.from_node_schema(
-        SemgrepGoLibrarySchema(), common_job_parameters,
-    )
-    go_libraries_cleanup_job.run(neo4j_session)
+    logger.info(f"Running Semgrep Dependencies cleanup job for {dependency_schema().label}.")
+    GraphJob.from_node_schema(dependency_schema(), common_job_parameters).run(neo4j_session)
 
 
 @timeit
 def sync_dependencies(
     neo4j_session: neo4j.Session,
     semgrep_app_token: str,
+    ecosystems_str: str,
     update_tag: int,
     common_job_parameters: Dict[str, Any],
 ) -> None:
 
     deployment_id = common_job_parameters.get("DEPLOYMENT_ID")
     if not deployment_id:
         logger.warning(
-            "Missing Semgrep deployment ID, ensure that sync_deployment() has been called."
+            "Missing Semgrep deployment ID, ensure that sync_deployment() has been called. "
             "Skipping Semgrep dependencies sync job.",
         )
         return
 
-    logger.info("Running Semgrep dependencies sync job.")
+    if not ecosystems_str:
+        logger.warning(
+            "Semgrep is not configured to import dependencies for any ecosystems, see docs to configure. "
+            "Skipping Semgrep dependencies sync job.",
+        )
+        return
+
+    # TODO: instead of erroring out, do we want to catch and log the error thrown for an invalid ecosystem flag,
+    # to let other semgrep modules run?
+    # seems like we want to error if the user provided a value but it's invalid
+    ecosystems = _parse_and_validate_semgrep_ecosystems(ecosystems_str)
 
-    # fetch and load dependencies for the Go ecosystem
-    raw_go_deps = get_dependencies(semgrep_app_token, deployment_id, ecosystems=["gomod"])
-    go_deps = transform_dependencies(raw_go_deps)
-    load_dependencies(neo4j_session, SemgrepGoLibrarySchema, go_deps, deployment_id, update_tag)
+    logger.info("Running Semgrep dependencies sync job.")
 
-    cleanup(neo4j_session, common_job_parameters)
+    for ecosystem in ecosystems:
+        schema = ECOSYSTEM_TO_SCHEMA[ecosystem]
+        raw_deps = get_dependencies(semgrep_app_token, deployment_id, ecosystem)
```

Comment:
Probably an early performance improvement, but what do you think about calling `get_dependencies` from all the ecosystems in a single call and then splitting the results via `transform_dependencies`?

Semgrep API can sometimes be unreliable and reducing the number of open HTTP connections might help.

Reply:
I considered that, it just seemed like it would make the code more complex, especially if we ever need to do ecosystem-specific handling for dependencies. But I guess we can cross that bridge when we reach it. I'll take a stab at rearranging the logic so that get_dependencies is only called once

Reply:
> Semgrep API can sometimes be unreliable and reducing the number of open HTTP connections might help.

Getting all the dependencies in a single function won't reduce the number of API calls needed though because it'll still do the same pagination across all ecoystems.

> especially if we ever need to do ecosystem-specific handling for dependencies

For that reason I'm a big fan of repeating ourselves and avoiding generics early on since we don't know yet if some of the data types will need special treatment.

I wouldn't be against splitting it out to sync_npm_dependencies(), sync_go_dependencies(), .. etc..

The general pattern is to have the sync function call a get+transform+load where the get is a fairly thin wrapper around the upstream API, the transform gets the data to a shape that makes sense for db ingestion, and load does its thing.

Reply:
I don't have a strong opinion, organize it in a way that makes sense to you as long as the tests pass and the structure roughly resembles get+transform+load